usaspending_api/download/v2/base_download_viewset.py

```diff
@@ -98,17 +100,36 @@ def get_download_response(self, file_name):
         else:
             file_path = self.s3_handler.get_simple_url(file_name=file_name)
 
-        # Add additional response elements that should be part of anything calling this function
-        response = {
-            "status": download_job.job_status.name,
-            "url": file_path,
-            "message": download_job.error_message,
-            "file_name": file_name,
-            # converting size from bytes to kilobytes if file_size isn't None
-            "total_size": download_job.file_size / 1000 if download_job.file_size else None,
-            "total_columns": download_job.number_of_columns,
-            "total_rows": download_job.number_of_rows,
-            "seconds_elapsed": download_job.seconds_elapsed(),
-        }
+        base_response = {"file_name": file_name, "file_url": file_path}
+        if is_download_status:
+            response = {
+                "status": download_job.job_status.name,
+                "message": download_job.error_message,
+                **base_response,
+                # converting size from bytes to kilobytes if file_size isn't None
+                "total_size": download_job.file_size / 1000 if download_job.file_size else None,
+                "total_columns": download_job.number_of_columns,
+                "total_rows": download_job.number_of_rows,
+                "seconds_elapsed": download_job.seconds_elapsed(),
+            }
+        else:
+            # Generate the status endpoint for the file
+            if settings.IS_LOCAL:
+                protocol = "http"
+                host = "localhost:8000"
+            elif settings.DOWNLOAD_ENV == "production":
+                protocol = "https"
+                host = "api.usaspending.gov"
+            else:
+                protocol = "https"
+                host = f"{settings.DOWNLOAD_ENV}-api.usaspending.gov"
+
+            status_url = f"{protocol}://{host}/api/v2/download/status?file_name={file_name}"
```

Comment:
Could we generalize this logic somewhere? This is pretty brittle logic...

Reply:
I agree I am not the biggest fan but did not see a much better way. This should only be used in the download endpoint so there is no need to break this out into a different function or create another item in settings.py similar to the DATA_DICTIONARY_DOWNLOAD_URL

Reply:
reuse isn't the only reason to use helper functions. Just having a private function in this class that returns status_url will drop cognitive complexity by 3 and shorten each function.